src/python/pants/backend/python/tasks/build_local_python_distributions.py

```diff
@@ -39,53 +43,73 @@ def product_types(cls):
       with self.invalidated(dist_targets,
                             fingerprint_strategy=DefaultFingerprintStrategy(),
                             invalidate_dependents=True) as invalidation_check:
+        interpreter = self.context.products.get_data(PythonInterpreter)
+
         for vt in invalidation_check.invalid_vts:
           if vt.target.dependencies:
             raise TargetDefinitionException(
               vt.target, 'The `dependencies` field is disallowed on `python_dist` targets. '
                          'List any 3rd party requirements in the install_requirements argument '
                          'of your setup function.'
             )
-          self._create_dist(vt.target, vt.results_dir)
+          self._create_dist(vt.target, vt.results_dir, interpreter)
 
         for vt in invalidation_check.all_vts:
           dist = self._get_whl_from_dir(os.path.join(vt.results_dir, 'dist'))
           req_lib_addr = Address.parse('{}__req_lib'.format(vt.target.address.spec))
           self._inject_synthetic_dist_requirements(dist, req_lib_addr)
           # Make any target that depends on the dist depend on the synthetic req_lib,
           # for downstream consumption.
-          for dependent in build_graph.dependents_of(vt.target.address):
-            build_graph.inject_dependency(dependent, req_lib_addr)
-
-  def _create_dist(self, dist_tgt, dist_target_dir):
-    """Create a .whl file for the specified python_distribution target."""
-    interpreter = self.context.products.get_data(PythonInterpreter)
+          for dependent in self.context.build_graph.dependents_of(vt.target.address):
+            self.context.build_graph.inject_dependency(dependent, req_lib_addr)
 
+  def _copy_sources(self, dist_tgt, dist_target_dir):
     # Copy sources and setup.py over to vt results directory for packaging.
     # NB: The directory structure of the destination directory needs to match 1:1
     # with the directory structure that setup.py expects.
-    for src_relative_to_target_base in dist_tgt.sources_relative_to_target_base():
+    all_sources = list(dist_tgt.sources_relative_to_target_base())
+    for src_relative_to_target_base in all_sources:
       src_rel_to_results_dir = os.path.join(dist_target_dir, src_relative_to_target_base)
       safe_mkdir(os.path.dirname(src_rel_to_results_dir))
       abs_src_path = os.path.join(get_buildroot(),
                                   dist_tgt.address.spec_path,
                                   src_relative_to_target_base)
       shutil.copyfile(abs_src_path, src_rel_to_results_dir)
-    # Build a whl using SetupPyRunner and return its absolute path.
-    setup_runner = SetupPyRunner(dist_target_dir, 'bdist_wheel', interpreter=interpreter)
-    setup_runner.run()
+
+  def _create_dist(self, dist_tgt, dist_target_dir, interpreter):
+    """Create a .whl file for the specified python_distribution target."""
+    self.context.log.debug("dist_target_dir: '{}'".format(dist_target_dir))
```

Comment:
This would likely be more useful if it included the target as well.... or could remove it.

Reply:
Removed, I remember adding this line many moons ago.